stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -152,6 +154,96 @@ class StockReserveRuleRemoval(models.Model):
         "When empty, any packaging can be removed.",
     )
 
+    TOLERANCE_LIMIT = [
+        ("no_tolerance", "No tolerance"),
+        ("upper_limit", "Upper Limit"),
+        ("lower_limit", "Lower Limit"),
+    ]
+
+    tolerance_requested_limit = fields.Selection(
+        selection=TOLERANCE_LIMIT,
+        string="Tolerance on",
+        help="Selecting a tolerance limit type"
+        "No tolerance: lot qty should be equal required"
+        "Upper Limit: lot with higher qty than required"
+        "Lower Limit: lot with lower qty than required",
+    )
+
+    tolerance_requested_computation = fields.Selection(
+        selection=[
+            ("percentage", "Percentage (%)"),
+            ("absolute", "Absolute Value"),
+        ],
+        string="Tolerance computation",
+    )
+
+    tolerance_requested_value = fields.Float(string="Tolerance value", default=0.0)
+
+    tolerance_display = fields.Char(
+        compute="_compute_tolerance_display", store=True, string="Tolerance"
+    )
+
+    @api.constrains("tolerance_requested_limit")
+    def _check_tolerance_requested_limit(self):
+        if self.filtered(
+            lambda r: (
+                r.tolerance_requested_limit == "no_tolerance"
+                and (
+                    r.tolerance_requested_computation != "percentage"
+                    or r.tolerance_requested_value != 0.0
+                )
+            )
+        ):
+            raise models.UserError(
+                _(
+                    '"No Tolerance" must be have values -'
+                    ' "Tolerance computation" with value Percentage (%)\' '
+                    "and \"Tolerance value\" with value '0.0'"
+                )
+            )
+
+    @api.onchange("tolerance_requested_limit")
+    def _onchange_tolerance_requested_limit(self):
+        if self.tolerance_requested_limit == "no_tolerance":
+            self.update(
+                {
+                    "tolerance_requested_computation": "percentage",
+                    "tolerance_requested_value": 0.0,
+                }
+            )
```

Comment:
I would drop this. You have tolerance_requested_limit that says what to do. Any value in the other parameters may stay.

Reply:
> I would drop this. You have tolerance_requested_limit that says what to do. Any value in the other parameters may stay.

Access rules to field only from view and without backend is not good solution. So I can't drop this.

Reply:
I thinks is not need remove this functions. In backend must be validate data to, not only in form view.

Reply:
But the rule will behave the same with or without those methods. You just loose the previous tolerance config when changing if want to apply tolerance or not. I'm not fan of this but this is just an opinion.